yarn-project/blob-sink/src/client/http.ts

```diff
@@ -412,35 +417,30 @@ async function getRelevantBlobs(
   logger: Logger,
   onBlobDeserializationError: 'warn' | 'trace' = 'warn',
 ): Promise<BlobWithIndex[]> {
-  const blobsPromise = data
-    // Filter out blobs not requested
-    .filter((b: BlobJson) => {
-      if (blobHashes.length === 0) {
-        return true;
-      }
-      const commitment = Buffer.from(b.kzg_commitment.slice(2), 'hex');
-      const blobHash = Blob.getEthVersionedBlobHash(commitment);
-      logger.trace(`Filtering blob with hash ${blobHash.toString('hex')}`);
-      return blobHashes.some(hash => hash.equals(blobHash));
-    })
-    // Attempt to deserialise the blob
-    // If we cannot decode it, then it is malicious and we should not use it
-    .map(async (b: BlobJson): Promise<BlobWithIndex | undefined> => {
-      try {
-        const blob = await Blob.fromJson(b);
-        return new BlobWithIndex(blob, parseInt(b.index));
-      } catch (err) {
-        if (err instanceof BlobDeserializationError) {
-          logger[onBlobDeserializationError](`Failed to deserialise blob`, { commitment: b.kzg_commitment });
-          return undefined;
-        }
-        throw err;
+  const hashToBlob = new Map<string, BlobJson>(
+    data.map(b => [bufferToHex(Blob.getEthVersionedBlobHash(hexToBuffer(b.kzg_commitment))), b] as const),
+  );
```

Comment:
IIUC this is where the magic happens. We overwrite the same map slot for blobs with the same kzg commitment